src/source/Dtls/Dtls.c

```diff
@@ -34,17 +34,37 @@ STATUS dtlsTransmissionTimerCallback(UINT32 timerID, UINT64 currentTime, UINT64
 {
     STATUS retStatus = STATUS_SUCCESS;
     PDtlsSession pDtlsSession = (PDtlsSession) customData;
+    BOOL locked = FALSE;
+    struct timeval timeout = {0};
+    UINT64 timeoutValDefaultTimeUnit = 0;
 
     CHK(pDtlsSession != NULL, STATUS_NULL_ARG);
 
+    MUTEX_LOCK(pDtlsSession->sslLock);
+    locked = TRUE;
+
     CHK(!SSL_is_init_finished(pDtlsSession->pSsl), STATUS_TIMER_QUEUE_STOP_SCHEDULING);
 
-    // do SSL_do_handshake to drive ssl state machine
-    SSL_do_handshake(pDtlsSession->pSsl);
-    CHK_STATUS(dtlsCheckOutgoingDataBuffer(pDtlsSession));
+    if (DTLSv1_get_timeout(pDtlsSession->pSsl, &timeout) == 0) {
```

Comment:
Possible to put `DTLSv1_get_timeout` directly inside `CHK` call?

Reply:
fix